pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +55,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = WrapClientErrors(err)
```

Comment:
Hmm...  I didn't think about that.  I think that the fact that our ClientConfig is used by other code that expects the existing set of errors makes it more important to avoid replacing them.  Any handling code written upstream will depend on their errors.  It would be like us changing errors on registries or clients.  Perhaps we could add an error printer as a parameter to the NewCommand calls.  

@smarterclayton suggestions?  It does seem like a quandary.